pkg/database/database.go

```diff
@@ -68,7 +68,13 @@ func NewMasterKeyAuthorizer(ctx context.Context, token azcore.TokenCredential, c
 		return nil, err
 	}
 
-	return cosmosdb.NewMasterKeyAuthorizer(*keys.PrimaryMasterKey)
+	return cosmosdb.NewMasterKeyAuthorizer(getDatabaseKey(keys, log))
+}
+
+func getDatabaseKey(keys sdkcosmos.DatabaseAccountsClientListKeysResponse, log *logrus.Entry) string {
+	keyName := "SecondaryMasterKey"
+	log.Infof("Using %s to authenticate with CosmosDB", keyName)
+	return *keys.SecondaryMasterKey
```

Comment:
Thinking a bit more about this, maybe it could be good to instead of passing the log here, this function could return both the key and the "contextMessage" and do the log after calling the function. In that way we can simplify the test case just comparing the expected values without needing to deal with assertions in logs (we also avoid touching I/O in the function that retrieves the key).
Something like:
```go
func getDatabaseKey(keysResponse sdkcosmos.DatabaseAccountsClientListKeysResponse) (key, contextInfo string) {
	key = *keysResponse.SecondaryMasterKey

	keyName := "SecondaryMasterKey"
	contextInfo = fmt.Sprintf("Using %s to authenticate with CosmosDB", keyName)

	return key, contextInfo
}
```
or even
```go
func getDatabaseKey(keysResponse sdkcosmos.DatabaseAccountsClientListKeysResponse) (key, contextInfo string) {
	return *keysResponse.SecondaryMasterKey, fmt.Sprintf("Using %s to authenticate with CosmosDB", "SecondaryMasterKey")
}
```
and in func NewMasterKeyAuthorizer() ... on line 71 do
```go
        key, contextInfo := getDatabaseKey(keys)
	log.Infof(contextInfo)

	return cosmosdb.NewMasterKeyAuthorizer(key)
```

Reply:
I'm going to leave it as since it's working and we'll drop all this in the near future.

Reply:
I'm +1 to both the suggested change as well as the decision to not pursue it within this PR.